kansha/card_addons/members/models.py

```diff
@@ -5,4 +5,52 @@
 # This software is licensed under the BSD License, as described in
 # the file LICENSE.txt, which you should have received as part of
 # this distribution.
-#--
\ No newline at end of file
+#--
+
+from nagare import database
+from sqlalchemy import func
+from elixir import ManyToOne, ManyToMany, using_options
+
+from kansha.models import Entity
+from kansha.card.models import DataCard
+from kansha.user.models import DataUser
+from kansha.column.models import DataColumn
+
+
+class DataMembership(Entity):
+    using_options(tablename='user_cards__card_members')
+
+    user = ManyToOne(DataUser, primary_key=True)
+    card = ManyToOne(DataCard, primary_key=True)
+
+    @classmethod
+    def get_for_card(cls, card):
+        return cls.query.filter_by(card=card)
+
+    @staticmethod
+    def favorites_for(card):
+        query = database.session.query(DataUser)
```

Comment:
The simplest would be to have another query I guess like: (not tested)

`def favorites_for(card):
    q2 = database.session.query(DataUser.id).join(DataMembership).filter(DataMembership.card == card)
    query = database.session.query(DataUser)
    query = query.join(DataMembership)
    query = query.join(DataCard).join(DataColumn).filter(DataColumn.board == card.column.board)
    query = query.filter(~DataUser.user_id.in_(q2))
    query = query.group_by(DataUser.username, DataUser.source)
    query = query.order_by(func.count(DataUser.username).desc())
    return query
`